src/main/kotlin/io/zerobase/smarttracing/resources/Email.kt

```diff
@@ -0,0 +1,99 @@
+package io.zerobase.smarttracing.resources
+
+import io.zerobase.smarttracing.models.*
+import javax.mail.Message
+import javax.mail.MessagingException
+import javax.mail.Session
+import javax.mail.internet.AddressException
+import javax.mail.internet.InternetAddress
+import javax.mail.internet.MimeMessage
+import javax.mail.internet.MimeMultipart
+import javax.mail.internet.MimeBodyPart
+import javax.mail.util.ByteArrayDataSource
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.util.Properties
+import java.io.File
+import java.io.InputStream
+import com.google.common.io.Resources
+import javax.activation.DataHandler
+
+// Currently only implements the AmazonSES
+class Email(private val fromEmailSES: String, private val amazonSES: AmazonSES,
+            private val config: EmailConfig) {
```

Comment:
a) don't include `SES` in variable names for other classes
b) the name of this class suggests that it represents an actual email object, but it seems like it's a wrapper for sending emails
c) the point of an abstraction is to use the abstraction. why are you passing in `AmazonSES` instead of `EmailSender`

Reply:
what name do you suggest for this email class? this class creates the email object and passes the raw byte array into the object that is sending the email

Reply:
the name of the class say this is an email, but that's not what it is. this is a utility class that sends emails. an email has to/from addresses, a subject, a body, and optionally an attachment. this class has configuration on how to send an email.

again, as with my comments from above, i think we should separate out the various parts of sending an email:
- rendering a body
- rendering an attachment
- sending the email

we want an interface for the last one, as discussed elsewhere in this PR, that takes the parts of an email and lets the implementation decide how to package them together. we want the business logic to decide if an attachment is necessary and call the relevant attachment generation code with the correct arguments (this is what emily is working on). 

for the rendering of the body, since each message could
a) be template and each template has the possibility of different arguments and
b) the same message could be rendered differently for different notification mediums (text, email, etc) 
i think it makes sense for each message type to implement its own rendering. a good option, IMO, would be use kotlin sealed classes for this. here's an example of what it could look like:

```kotlin
sealed class UserNotification {
    abstract fun render(medium: NotificationMedium = NotificationMedium.EMAIL): String
}

class OrganizationOnboardingNotification(val organizationName: String): UserNotification {
    private val emailTemplate = "notifications/organization-onboarding/email.mustache"
    private val smsTemplate = "notifications/organization-onboarding/sms.mustache"

    override fun render(medium: NotificationMedium): String {
        // do the rendering
    }
}
```